src/Login/index.test.js

```diff
@@ -1,15 +1,75 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { BrowserRouter as Router } from 'react-router-dom';
+import { MemoryRouter as Router, Switch, Route } from 'react-router-dom';
+import { MockedProvider } from '@apollo/client/testing';
+import AuthProvider from '../AuthProvider';
 import Login from './index';
+import QUERY_USERS from '../graphql/queries';
+import { HOME_PATH, LOGIN_PATH } from '../paths';
 
-it('renders the login page', () => {
-  render(<Router><Login /></Router>);
+const mockedUsersResponse = {
+  request: { query: QUERY_USERS },
+  result: {
+    data: {
+      users: [
+        {
+          username: 'User',
+          password: 'password',
+          id: 1,
+        },
+      ],
+    },
+  },
+};
+
+const mocks = [
+  mockedUsersResponse,
+  mockedUsersResponse,
+  mockedUsersResponse,
```

Comment:
I'm not sure why you needed to duplicate `mockedUsersResponse` here 3 times, but if it's to have 3 users in the array then you should add 2 more users with different properties in `mockedUsersResponse`, not sure if it would work like this but let me know if you have issues with it, ideally we want to have something more realistic like this:
```
      users: [
        {
          username: 'Danny',
          password: '632569',
          id: 1,
        },
        {
          username: 'James',
          password: '12345',
          id: 2,
        },
        {
          username: 'William',
          password: '678912',
          id: 3,
        }
      ]
```

Reply:
The MockedProvider component requires separate mock response objects for each query, even if the query is being repeated.

Reply:
Though it's unnecessary anyway now after splitting up the tests into multiple cases.

Reply:
Ok, but would there be an issue with having multiple users in the response rather than just one? My main goal is to mimic a real situation where we test that this code would be solid when we are searching for a user that's part of a large collections of existing users: `result.data.users.find((u) => u.username === username && u.password === password);`
But if we only have one user in the mock, our test won't be as efficient I think, does it make sense?